File: 0_팀노트/소수/prime.py
def is_prime(n):
    if n <= 1:
      return False
    i = 2
    while i * i <= n:
        if n % i == 0: 
          return False
        i += 1
    return True

# n은 범위
def make_primes(n:int)->list:
    primes = [True] * (n + 1)
    primes[0:2] = [False, False]
    for num in range(2, int(n * 0.5) + 1):
        if primes[num]:
            for i in range(num ** 2, n + 1, num):
                primes[i] = False
    return primes

File: 0_팀노트/소수/prime.js
// 에라토스테네스의 체
// n은 범위
function makePrimes(n) {
  const primes = Array.from({ length: n + 1 }, (v) => true)
  primes.splice(0, 2, false, false)
  for (let num = 2; num < Math.floor(Math.sqrt(n)) + 1; num++) {
    if (primes[num]) {
      for (let i = num * num; i < n + 1; i += num) {
        primes[i] = false
      }
    }
  }
  return primes
}

console.log(makePrimes(20)) // [false, false, true, true, false ....]

// 에라토스테네스의 체가 아닌 소수판별 알고리즘
function isPrime(number) {
  if (number <= 1) {
    return false
  }
  let i = 2
  while (i * i <= number) {
    if (number % i === 0) {
      return false
    }
    i++
  }
  return true
}
